tests/unit/internal-session-test.js

```diff
@@ -189,6 +196,38 @@ describe('InternalSession', () => {
     describe('when the restored data does not contain an authenticator factory', () => {
       itDoesNotRestore();
     });
+
+    describe('with older synchronous stores (< v1.1.0)', function() {
+      describe('when restoring the session', function() {
+        describe('when the store resolves restoration', function() {
+          beforeEach(() => {
+            sinon.stub(store, 'restore').returns({ authenticated: { authenticator: 'authenticator' } });
+            sinon.stub(store, 'persist').returns();
+            sinon.stub(store, 'clear').returns();
+          });
+
+          it('is authenticated', () => {
+            return session.restore().then(() => {
+              expect(session.get('isAuthenticated')).to.be.true;
+            });
+          });
+        });
+
+        describe('when the store rejects restoration', function() {
```

Comment:
If the stores now have a promise based API we need to handle rejection as well. I don't really know what kind of stores people might be using but if persisting rejects then session authentication should reject as well. If that's not correctly handled, people might see strange behavior in case their custom stores reject persisting the session.

As Ember Simple Auth is a library with extension points (e.g. custom stores) it must implement all scenarios completely, e.g. in this case is must also handle rejected promises if it expects promises to be returned from the session stores.